devito/passes/clusters/aliases.py

```diff
@@ -1,15 +1,15 @@
-from collections import OrderedDict, namedtuple
+from collections import OrderedDict, defaultdict
```

Comment:
@mloubout basically this file is what this newer PR is changing wrt #1154 

it's a pretty big rewrite, because I'm generalising loads of concepts and objects. But it's a very (very) nice cleanup, I think